lib/watir/locators/element/locator.rb

```diff
@@ -49,53 +45,31 @@ def locate_all
         private
 
         def using_selenium(filter = :first)
-          selector = @selector.dup
-          tag_name = selector[:tag_name].is_a?(::Symbol) ? selector[:tag_name].to_s : selector[:tag_name]
-          selector.delete(:tag_name) if selector.size > 1
-
-          WD_FINDERS.each do |sel|
-            next unless (value = selector.delete(sel))
-            return unless selector.empty? && wd_supported?(sel, value)
-            if filter == :all
-              found = locate_elements(sel, value)
-              return found if sel == :tag_name
-              filter_selector = tag_name ? {tag_name: tag_name} : {}
-              return filter_elements(found, filter_selector, filter: filter).compact
-            else
-              found = locate_element(sel, value)
-              return sel != :tag_name && tag_name && !validate([found], tag_name) ? nil : found
-            end
-          end
-          nil
+          return unless @selector.size == 1
```

Comment:
This doesn't work for `:link`, `:link_text` and `:partial_link_text - eg:

~~~~~~~~ruby
browser.link(link_text: 'valid_text_value').text
~~~~~~~~

The `@selector` will be `{:link_text=>'valid_text_value', :tag_name=>'a'}`. It'll get pushed to `using_watir`, which treats this as an attribute instead of Seleniums special link logic.

Reply:
Ah, right, well, this will be more simple when we just stop supporting them. :-D